python/demo/main.py

```diff
@@ -142,40 +144,40 @@ class GuestbookPost(db.Model):
   write_time = db.DateTimeProperty(auto_now_add=True)
 
 
-class StartPipelineHandler(webapp.RequestHandler):
+class StartPipelineHandler(MethodView):
   def get(self):
-    self.response.out.write(template.render('start.html', {}))
+    return render_template('start.html')
 
   def post(self):
-    colors = [color for color in self.request.get_all('color') if color]
+    colors = [color for color in request.form.getlist('color') if color]
     job = CountReport(
         users.get_current_user().email(),
         GuestbookPost.kind(),
         'color',
         *colors)
     job.start()
-    self.redirect('/_ah/pipeline/status?root=%s' % job.pipeline_id)
+    return redirect('/_ah/pipeline/status?root=%s' % job.pipeline_id)
 
 
-class MainHandler(webapp.RequestHandler):
+class MainHandler(MethodView):
   def get(self):
-    context = {'posts': GuestbookPost.all().order('-write_time').fetch(100)}
-    self.response.out.write(template.render('guestbook.html', context))
+    posts = GuestbookPost.all().order('-write_time').fetch(100)
+    return render_template('guestbook.html', posts=posts)
 
   def post(self):
-    color = self.request.get('color')
+    color = request.form.get('color')
     if color:
       GuestbookPost(color=color.lower()).put()
-    self.redirect('/')
+    return redirect('/')
+
 
+app = Flask(__name__)
 
-def main():
-  application = webapp.WSGIApplication([
-      (r'/', MainHandler),
-      (r'/pipeline', StartPipelineHandler),
-  ], debug=True)
-  util.run_wsgi_app(application)
+app.wsgi_app = wrap_wsgi_app(app.wsgi_app, use_legacy_context_mode=True)
 
+app.add_url_rule('/', view_func=MainHandler.as_view('main'))
+app.add_url_rule('/pipeline', view_func=StartPipelineHandler.as_view('pipeline'))
 
-if __name__ == '__main__':
-    main()
```

Comment:
Reminder for myself: expose some kind of function or middleware for apps to call to register routes like below.